shared-libs/cht-datasource/src/libs/core.ts

```diff
@@ -33,13 +33,14 @@ const isDataPrimitive = (value: unknown): value is DataPrimitive => {
     || value instanceof Date;
 };
 
-interface DataArray extends Readonly<DataValue[]> { }
+type DataArray = readonly DataValue[]
 
 const isDataArray = (value: unknown): value is DataArray => {
   return Array.isArray(value) && value.every(v => isDataPrimitive(v) || isDataArray(v) || isDataObject(v));
 };
 
 /** @internal */
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
```

Comment:
I cannot convert this interface to a type alias because TS does not let you have recursive type aliases (but you can do that with an interface).  

This `DataObject` interface is recursive because DataObject > DataValue > DataObject.